core/storage/feedback/gae_models.py

```diff
@@ -304,6 +306,42 @@ def get_messages(cls, exploration_id, thread_id):
             cls.thread_id == full_thread_id).fetch(feconf.DEFAULT_QUERY_LIMIT)
 
     @classmethod
+    def get_last_two_message_ids_of_threads(cls, exploration_ids, thread_ids):
+        """Returns a list of the last two messages of the given threads.
+
+        Args:
+            exploration_id: str. ID of the exploration the thread
+                belongs to.
+            thread_id: str. ID of the thread.
+
+        Returns:
+            list(int). A list of message ids present in the given thread, having
+                a maximum length of feconf.DEFAULT_QUERY_LIMIT.
+        """
+        message_ids = []
+        full_thread_ids = (
+            [FeedbackThreadModel.generate_full_thread_id(
+                exploration_id, thread_id)
+             for exploration_id, thread_id in zip(exploration_ids, thread_ids)])
+        thread_models = FeedbackThreadModel.get_multi(full_thread_ids)
```

Comment:
There shouldn't be get_multi() of a different model in here. That should be in the logic/services layer.

That said, please see my comment above regarding this. I think it could be a method of the FeedbackThread domain object.